crates/bevy_ecs/src/archetype.rs

```diff
@@ -124,47 +126,91 @@ pub struct Archetype {
     entities: Vec<Entity>,
     edges: Edges,
     table_info: TableInfo,
-    table_components: Cow<'static, [ComponentId]>,
-    sparse_set_components: Cow<'static, [ComponentId]>,
-    pub(crate) unique_components: SparseSet<ComponentId, Column>,
-    pub(crate) components: SparseSet<ComponentId, ArchetypeComponentInfo>,
+    table_components: Cow<'static, [(RelationKindId, Option<Entity>)]>,
```

Comment:
I think `Entity` should get a niche by for example making `generation` a `NonZeroU32`. This would make `Option<Entity>` just as big as `Entity`.

Reply:
I actually want to remove all the `Option<Entity>` everywhere and just have a dedicated entity for the `None` case as it'll make a lot of the API nicer. It's kind of annoying to have to return or take `Option<Entity>` in places where 99% of the time it will be `Some(..)`

Reply:
For example when setting the target filters on a Relation<T> query param we currently just take an `Entity` but really we ought to be taking an `Option<Entity>`. Problem then is 99% of use cases have to wrap it in Some just because someone might want to query for the `None`case... 

I could take some `T: Into<Option<Entity>>` for that *specific* function but there are other places where this kind of pattern doesnt work, like the return type of the `RelationAccess<T>` iterator that returns the target entity- except the target Entity should actually be `Option<T>` but if I did that you'd either be unwrapping or doing a bunch of boilerplate to handle the `None` case :S